File: backend/templates/database/repl/rust/src/repl.rs
use std::io::{self, Write};

pub fn run_repl() {
    let stdin = io::stdin();
    let mut input = String::new();
    loop {
        print!("db > ");
        io::stdout().flush().unwrap();
        input.clear();
        if stdin.read_line(&mut input).unwrap() == 0 {
            break;
        }
        let trimmed = input.trim_end();
        if trimmed == ".exit" {
            break;
        }
        println!("Unrecognized command '{}'.", trimmed);
    }
}

File: backend/templates/database/repl/python/repl.py
import sys


def run_repl():
    while True:
        print("db > ", end="", flush=True)
        try:
            line = input()
        except EOFError:
            break
        line = line.strip()
        if line == ".exit":
            break
        print(f"Unrecognized command '{line}'.")
